### ISSUE TITLE
n8n OAuth2 callback crashes when retrieving token from Reddit

### ISSUE THREAD

**Issue by joebobmiles** [n8n-io/n8n#1906]
**Describe the bug**
When getting OAuth2 access tokens from Reddit, n8n's callback URL crashes due to an unhandled promise rejection error.

Example request Reddit makes to the callback URL:
```
https://n8n.example.com/rest/oauth2-credential/callback?state=eyJ0b2tlbiI6IkpFdVhGUHM4LTdnRnNZTi1rLUhlSnVucXk2R0NaRHNWNEVLcyIsImNpZCI6IjYifQ%3D%3D&code=8yJ4Vy34MP2CsnXcWH9ixRsCfQscBA#_
```

Error message thrown by n8n:
```
n8n    | (node:6) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 15)
n8n    | (node:6) UnhandledPromiseRejectionWarning: Error: 401
n8n    |     at getAuthError (/usr/local/lib/node_modules/n8n/node_modules/client-oauth2/src/client-oauth2.js:122:15)
n8n    |     at /usr/local/lib/node_modules/n8n/node_modules/client-oauth2/src/client-oauth2.js:276:21
n8n    |     at runMicrotasks (<anonymous>)
n8n    |     at processTicksAndRejections (internal/process/task_queues.js:93:5)
n8n    |     at async /usr/local/lib/node_modules/n8n/dist/src/Server.js:876:32
```

**To Reproduce**
1. Install n8n via Docker, with Basic Auth enabled
2. Create a Reddit node for creating posts
3. Create a new credential for Reddit
4. Follow [documented process for adding Reddit credentials](https://docs.n8n.io/credentials/reddit/#using-oauth)

**Expected behavior**
Expected Reddit credentials to be retrieved after approving app access via Reddit's app access approval page.

**Environment (please complete the following information):**
 - OS: Alpine Linux
 - n8n Version: 0.118.1 (Docker image `n8nio/n8n`)
 - Node.js Version: 14.15.5 (via `docker run node --version`)
 - Database system: SQLite
 - Operation mode: `own`

**Additional context**

This n8n instance is running inside a Docker container, managed by Docker Compose and behind an Nginx reverse proxy. The reverse proxy simply `proxy_pass`s all requests made to the n8n subdomain to the local port where n8n is listening. Here's the (slightly redacted) `docker-compose.yml` and `.env` I'm using:

`docker-compose.yml`
```yml
services:
        n8n:
                container_name: n8n
                image: n8nio/n8n
                restart: unless-stopped
                ports:
                        - "${N8N_HOST_PORT}:${N8N_GUEST_PORT}"
                environment:
                        - N8N_BASIC_AUTH_ACTIVE=true
                        - N8N_BASIC_AUTH_USER=${N8N_USERNAME}
                        - N8N_BASIC_AUTH_PASSWORD=${N8N_PASSWORD}
                        - N8N_HOST
                        - N8N_PORT=${N8N_GUEST_PORT}
                        - N8N_PROTOCOL=https
                        - WEBHOOK_TUNNEL_URL=https://${N8N_HOST}/
                        - VUE_APP_URL_BASE_API=https://${N8N_HOST}/
                        - GENERIC_TIMEZONE=${N8N_TIMEZONE}
                        - NODE_ENV=production
                volumes:
                        - ${N8N_DATA_FOLDER}:/home/node/.n8n
```

`.env`
```env
N8N_HOST_PORT=127.0.0.1:8081
N8N_GUEST_PORT=8081

N8N_HOST=n8n.example.com

N8N_USERNAME=username
N8N_PASSWORD=password

N8N_TIMEZONE=America/Los_Angeles
N8N_DATA_FOLDER=/srv/services/n8n/data
```

**Comment by janober** [comment]
Thanks for reporting. The issue with the uncaught error got fixed and will be released with the next version. I will update here once it got released (probably tomorrow).

Sadly do not know why you get this error in the first place as I was sadly not able to reproduce it and works totally fine for me on my server with a similar setup (not using nginx rather traefik). So do currently not expect it to be an issue in n8n as additionally nobody else ever reported an issue like that before and no changes regarding OAuth2 have been made in a very long time.

So please test this again and also test with other OAuth2 applications. You could also try again following the official [Server Setup guide](https://docs.n8n.io/getting-started/installation/advanced/server-setup.html) and if you there face the same issue, to be sure that it is not nginx-setup related.

I close this issue for now but please report back in both cases. If you get it working and also if you do not get it working that I can open the issue again. Thanks!

**Comment by janober** [comment]
Got released with n8n@0.130.0
